Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -178,5 +178,257 @@ public void DoGetRequest()
+    ///   <code>
+    /// PS C:\> $job = Start-BqJob -Query "select * from classics where Year > 1900" `
+    /// -DefaultDataset $dataset -DestinationTable $table
+    ///   </code>
+    ///   <para>Queries with a default dataset and using a permanent table as the destination for results.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs)">
+    /// [BigQuery Jobs]
+    /// </para>
+    /// </summary>
+    [Cmdlet("Start", "BqJob")]
+    public class StartBqJob : BqCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string DoQuery = "DoQuery";
+            public const string DoCopy = "DoCopy";
+            public const string DoLoad = "DoLoad";
+            public const string DoExtract = "DoExtract";
+        }
+
+        // All-Type Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for jobs in. If not set via PowerShell parameter processing, it will
+        /// default to the Cloud SDK's default project.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public override string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Turns the async call into a synchronous call by polling until the job is complete before 
+        /// returning. Can also be accessed by '-Synch'.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [Alias("Synch")]
+        public SwitchParameter PollUntilComplete { get; set; }
+
+        // Query Parameters.
+        //TODO(ahandley): Billing params for Queries.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Query.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoQuery)]
+        public SwitchParameter Query { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// A query string, following the BigQuery query syntax, of the query to execute.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.DoQuery)]
+        [ValidateNotNullOrEmpty]
+        public string QueryString { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Causes BigQuery to not run the job. Instead, if the query is valid, BigQuery returns 
+        /// statistics about the job such as how many bytes would be processed. If the query is 
+        /// invalid, an error returns. //This feature is not currently available.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        public SwitchParameter DryRun { get; set; }
```

Comment:
We should probably make use of the built-in -WhatIf parameter of PowerShell (when you do SupportsShouldProcess) and remove this parameter.